Description: Re-add exa hang workaround.
 .
 This reverts commit 321eb2df5c8b1da4d6469037dd1d8888b508fabd:
 .
    Revert "nv04-nv40/exa: Match the blob behavior more closely on PrepareCopy."
 .
    This reverts commit 9de0d97bd2fc2ee8800d48b5340a3d495525ad3e. Unnecessary
    after kernel commit "drm/nouveau: Ack the context switch interrupt before
    switching contexts.".
 .
 We do not yet have the named commit in our kernel tree.


Index: xserver-xorg-video-nouveau/src/nv04_exa.c
===================================================================
--- xserver-xorg-video-nouveau.orig/src/nv04_exa.c	2010-08-05 15:03:35.247880586 +1000
+++ xserver-xorg-video-nouveau/src/nv04_exa.c	2010-08-05 15:04:17.969451030 +1000
@@ -207,11 +207,15 @@
 			return FALSE;
 		}
 
+		BEGIN_RING(chan, blit, NV04_IMAGE_BLIT_SURFACE, 1);
+		OUT_RING  (chan, pNv->NvContextSurfaces->handle);
 		BEGIN_RING(chan, blit, NV01_IMAGE_BLIT_OPERATION, 1);
 		OUT_RING  (chan, 1); /* ROP_AND */
 
 		NV04EXASetROP(pScrn, alu, planemask);
 	} else {
+		BEGIN_RING(chan, blit, NV04_IMAGE_BLIT_SURFACE, 1);
+		OUT_RING  (chan, pNv->NvContextSurfaces->handle);
 		BEGIN_RING(chan, blit, NV01_IMAGE_BLIT_OPERATION, 1);
 		OUT_RING  (chan, 3); /* SRCCOPY */
 	}
